onnxruntime/core/providers/coreml/coreml_execution_provider.cc

```diff
@@ -36,6 +37,163 @@ CoreMLExecutionProvider::CoreMLExecutionProvider(const CoreMLOptions& options)
 
 CoreMLExecutionProvider::~CoreMLExecutionProvider() {}
 
+// Create a helper function for type checking
+bool CoreMLExecutionProvider::IsSupportedInputOrOutputType(int32_t type) const {
+  // Supported input and output types for CoreML MLProgram EP
```

Comment:
does this limitation only apply for MLProgram? if so, should we limit the partition filtering to that case?